### PR TITLE
Magma device inertia

### PR DESCRIPTION
**Pull request by cnpetra**
Addresses robustness performance aspects related to using MAGMA on device

* inertia calculation dsidi with Bunch-Kaufman solver
* inertia calculation on device with nopiv solver

Requires MAGMA 2.5.4 (uses API introduced in this version)

To fully test this PR, one needs to use option `linsol_mode speculative`, which is a non-default option (CI pipelines do not cover it). This option was tested manually on summit.

### CODE REVIEW

**Review by ashermancinelli — APPROVED** [review]
I'll try to get this running with the newly installed magma ver on ascent. Also I think this may need a rebase before merging.

**Review by pelesh — CHANGES_REQUESTED** [review]
This looks good. Perhaps we should make sure Magma 2.5.4 is installed on all CI pipelines before merging this?

File: BUILD.sh [hunk 1]
```diff
@@ -118,11 +118,12 @@ ascent)
     module purge
     module load cuda/11
     module use /gpfs/wolf/proj-shared/csc359/ascent/Modulefiles/Core
+    module use /gpfs/wolf/proj-shared/csc359/ascent/spack-modulefiles/
     module load exasgd-base
     module load gcc-ext/7.4.0
     module load spectrum-mpi-ext
     module load openblas
-    module load magma/2.5.3-cuda11
+    module load magma-2.5.4-gcc-7.4.0-vjotnd3 #magma/2.5.3-cuda11
```
[the remark this answers is not in the document]
Comment (ashermancinelli) [reply BUILD.sh:126]: We have it on all but marianas, I'm working on it now.

**Comment by ashermancinelli** [comment]
I've added magma 2.5.4 builds to all target clusters, and after #169 is merged and this PR is rebased on develop I think it's ready. CC @pelesh

**Review by pelesh — CHANGES_REQUESTED** [review]
Once #169 is merged, this will be ready to merge, as well.

**Comment by ashermancinelli** [comment]
> Once #169 is merged, this will be ready to merge, as well.
> 
> CC @cnpetra

Does this need rebase?

**Review by pelesh — APPROVED** [review]
Ready to merge!

**Pull request merged**